readthedocs/search/parse_json.py

```diff
@@ -195,8 +205,8 @@ def parse_content(content, remove_first_line=False):
     if remove_first_line and len(content) > 1:
         content = content[1:]
 
-    # converting newlines to ". "
-    content = ' '.join(text.strip() for text in content if text)
+    content = map(lambda x: x.strip(), content)
+    content = ' '.join(text for text in content if text)
```

Comment:
I don't love functional code like this. It seems really complex for what it's doing. We should just write this section as a loop to make it more explicit.